src/vanilla.ts

```diff
@@ -116,14 +116,24 @@ const buildProxyFunction = (
       const desc: PropertyDescriptor = {
         value,
         enumerable: true,
+        // This is intentional to avoid copying with proxy-compare.
+        // It's still non-writable, so it avoids assigning a value.
+        configurable: true,
```

Comment:
This allows to change the configuration of the property, but it's probably not often done.
By the way, we already removed Object.freeze in #656, so assigning a new property was possible.
After all, non-writable is just to avoid misusage of snapshots.

Reply:
Ah right; shoot--even though this was not doing an `Object.freeze`, the `configurable: false` + `writable: false` defaults kicked in the "I'm going to copy now" behavior.

Just wondering, I was going to ask in the proxy-compare PR, but what do you think about, instead of copying, just failing the proxy-compare `createProxy` call if we detect non-configureable+non-writeable properties, and ask the caller to fix their properties (either by doing a copy on their side, or adjusting their descriptors)?

I'm thinking that making the copy might be too implicit/too magical, and maybe most callers would be like this code, which didn't realize it was asking `createProxy` to do a copy, and it was actually be pretty easy to throw in the "oh sure, `configurable: true`" on the caller end?

Reply:
Also, wdyt of adding a unit test that covers `snapshot`s properties are configurable? 

...or maybe make a snapshot, make the compare-proxy, call `getUntracked` and make sure it's the same instance as the snapshot? Oh, except `getUntracked` is smart enough to, when a copy is made, it still returns the non-copy anyway.

I liked the assert-again-`getUntracked` idea b/c it would be more black-box and assert what we really care about (compare-proxy isn't making copies), but that's fine, maybe just asserting against the descriptor is good enough.

...or maybe call compare-proxy with a `targetCache` and assert the map stays empty, which would imply no copies were made.